compiler/jsgen.nim

```diff
@@ -2733,20 +2733,13 @@ proc genCast(p: PProc, n: PNode, r: var TCompRes) =
   elif toUint and src.kind in {tyInt64, tyUInt64} and optJsBigInt64 in p.config.globalOptions:
     r.res = "Number(BigInt.asUintN($1, $2))" % [$(dest.size * 8), r.res]
   elif toInt:
-    if fromInt:
-      return
-    elif fromUint:
-      if src.size == 4 and dest.size == 4:
-        # XXX prevent multi evaluations
-        r.res = "($1 | 0)" % [r.res]
+    if fromInt or fromUint:
+      if dest.size == 4:
+        r.res = "(($1) | 0)" % [r.res]
```

Comment:
I added parentheses around `$1`, not sure if that's necessary, but it's also done in `genConv`. I also removed the comment about multi evaluations, as I don't see how that could happen here.